apps/code-examples/src/app/code-examples/modals/modal/mock-modal-svc.component.spec.ts

```diff
@@ -0,0 +1,75 @@
+/**
+ * Note: This spec demonstrates how to mock out the `SkyModalService`
+ * and spy on calls to its methods.
+ */
+import { Component, inject } from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { SkyModalInstance, SkyModalService } from '@skyux/modals';
+import { SkyModalTestingModule } from '@skyux/modals/testing';
+
+import { ModalComponent } from './modal.component';
+
+@Component({
+  standalone: true,
+  template: `<button
+    class="sky-btn"
+    data-sky-id="my-modal-open-button"
+    type="button"
+    (click)="edit()"
+  >
+    Edit...
+  </button>`,
+})
+class TestComponent {
+  public modalClosedReason: string | undefined;
+
+  readonly #modalSvc = inject(SkyModalService);
+
+  protected edit(): void {
+    const instance = this.#modalSvc.open(ModalComponent);
+
+    instance.closed.subscribe((args) => {
+      this.modalClosedReason = args.reason;
+    });
+  }
+}
+
+describe('Modal service', () => {
+  function setupTest(): {
+    fixture: ComponentFixture<TestComponent>;
+    modalSvc: SkyModalService;
+  } {
+    const fixture = TestBed.createComponent(TestComponent);
+    const modalSvc = TestBed.inject(SkyModalService);
+
+    return { fixture, modalSvc };
+  }
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [SkyModalTestingModule, TestComponent],
+    });
+  });
+
+  it('should open and close a modal', () => {
+    const { fixture, modalSvc } = setupTest();
+
+    const modalInstance = new SkyModalInstance();
+    const openSpy = spyOn(modalSvc, 'open').and.returnValue(modalInstance);
```

Comment:
Using Jasmine spies to validate when a modal is opened seems odd to me. The typical pattern for Angular test modules is to provide a testing controller to validate that certain actions take place. See Angular's `[HttpTestingController](https://angular.io/api/common/http/testing/HttpTestingController)` for an example.

I could imagine the test controller having methods like `expectOpen()`, `close()` that provides `reason` and `data` values, etc.